runtime/context.go

```diff
@@ -99,6 +100,25 @@ func AnnotateIncomingContext(ctx context.Context, mux *ServeMux, req *http.Reque
 	return metadata.NewIncomingContext(ctx, md), nil
 }
 
+func isValidGRPCMetadataKey(key string) bool {
+	for i := 0; i < len(key); i++ {
+		r := key[i]
```

Comment:
There should be no reason to not just iterate over each rune in this slice, right?
```suggestion
	for i, r := range key {
```

Reply:
Done now, right?